server/project.go

```diff
@@ -75,26 +75,41 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 	g.GET("/project", func(c echo.Context) error {
 		ctx := c.Request().Context()
 		projectFind := &api.ProjectFind{}
+		var principalID *int
 		if userIDStr := c.QueryParam("user"); userIDStr != "" {
 			userID, err := strconv.Atoi(userIDStr)
 			if err != nil {
 				return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Query parameter user is not a number: %s", userIDStr)).SetInternal(err)
 			}
-			projectFind.PrincipalID = &userID
+			principalID = &userID
 		}
 
 		// Only Owner and DBA can fetch all projects from all users.
-		if projectFind.PrincipalID == nil {
+		if principalID == nil {
 			role := c.Get(getRoleContextKey()).(api.Role)
 			if role != api.Owner && role != api.DBA {
 				return echo.NewHTTPError(http.StatusForbidden, "Not allowed to fetch all project list")
 			}
+		} else {
+			member, err := s.store.GetMemberByPrincipalID(ctx, *principalID)
```

Comment:
Shall we move this block of code to line 83 because that's when principalID or projectFind.PrincipalID != nil?

Reply:
I guess you just want to make it closer to lin 83. So I move line 83 near it. 😆